### PR TITLE
Refactor generator

### LINKED ISSUE #4221: Fix tmp dir strategy for hypothesis config generation
When running tests, we often employ the tmp dir pattern from pytest, in which we create a separate new temporary directory that is used for the duration of the test, and left again after the test was run. this gives separation of tests that do file system IO in CWD.

hypothesis hacks our tests, and runs the test function over and over with its different generated examples.

sadly the hypothesis hack and the pytest tmp dir pattern don't play together well - at the moment, the way its written, hypothesis runs all its examples in the same temp dir. that is unfortunate, because we have tests that create files in CWD.

we hope to find a way to create a new tmp dir for each separate example coming from hypothesis.

### PR DESCRIPTION
**Pull request by mortalisk**
**Issue**
Resolves #4221 

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -37,17 +42,21 @@ jobs:
       run: ./script/clang-format --check
 
     - name: CMake Format
+      if: ${{ always() }}
```
Comment (eivindjahren) [inline .github/workflows/style.yml:45]: What does this do?

File: src/ert/_c_wrappers/enkf/model_config.py [hunk 2]
```diff
@@ -33,15 +31,8 @@ def __init__(
...
-            config_dir = os.getcwd()
-        self._config_dir = config_dir
-        self.data_root = data_root if data_root is not None else config_dir
-        self.ens_path: str = (
-            ens_path
-            if ens_path is not None
-            else os.path.join(config_dir, self.DEFAULT_ENSPATH)
-        )
+        self.data_root = data_root
+        self.ens_path: str = ens_path
```
Comment (eivindjahren) [inline src/ert/_c_wrappers/enkf/model_config.py:35]: Typing issue here.

File: .github/workflows/style.yml (hunk 1 above)
Comment (mortalisk) [reply .github/workflows/style.yml:45]: Runs even if the thing fails. Perhaps it should be a separate PR witth these ci changes?

Comment (mortalisk) [reply .github/workflows/style.yml:45]: Or, perhaps you questioned the sanity of doing it to the first check? If so, I have no answer

File: tests/test_config_parsing/test_ensemble_config.py [hunk 3]
```diff
@@ -4,62 +4,72 @@
...
-@given(config_dicts())
-def test_ensemble_config_from_file_and_dict_coincide(config_dict):
+@given(config_generators())
+def test_ensemble_config_from_file_and_dict_coincide(
+    tmp_path_factory, config_generator
+):
     filename = "config.ert"
-    to_config_file(filename, config_dict)
+    with config_generator(tmp_path_factory, filename) as config_dict:
+        to_config_file(filename, config_dict)
```
Comment (eivindjahren) [inline tests/test_config_parsing/test_ensemble_config.py:16]: Perhaps the contextmanager should perform `to_config_file` for you ?

**Review by eivindjahren — CHANGES_REQUESTED** [review]
I mainly just had questions.

File: tests/test_config_parsing/test_ensemble_config.py (hunk 3 above)
Comment (mortalisk) [reply tests/test_config_parsing/test_ensemble_config.py:16]: :100:  true

File: .github/workflows/style.yml (hunk 1 above)
Comment (eivindjahren) [reply .github/workflows/style.yml:45]: Could you separate it into its own commit?

**Comment by mortalisk** [comment]
Jenkins test ert please

**Comment by eivindjahren** [comment]
Needs rebasing.

**Pull request merged**